src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisResultService.java

```diff
@@ -72,34 +83,21 @@ public SecurityAnalysisResultService(SecurityAnalysisResultRepository securityAn
     public List<PreContingencyLimitViolationResultDTO> findNResult(UUID resultUuid, List<ResourceFilterDTO> resourceFilters, Sort sort) {
         assertResultExists(resultUuid);
         assertPreContingenciesSortAllowed(sort);
-        Specification<PreContingencyLimitViolationEntity> specification = preContingencyLimitViolationRepository.getParentsSpecifications(resultUuid, resourceFilters);
-        Sort newSort = createNResultSort(sort);
-        List<PreContingencyLimitViolationEntity> preContingencyLimitViolation = preContingencyLimitViolationRepository.findAll(specification, newSort);
+        Specification<PreContingencyLimitViolationEntity> specification = preContingencyLimitViolationSpecificationBuilder.buildSpecification(resultUuid, resourceFilters);
+
+        List<PreContingencyLimitViolationEntity> preContingencyLimitViolation = preContingencyLimitViolationRepository.findAll(specification, sort);
         return preContingencyLimitViolation.stream()
                 .map(PreContingencyLimitViolationResultDTO::toDto)
                 .toList();
     }
 
     @Transactional(readOnly = true)
     public byte[] findNResultZippedCsv(UUID resultUuid, CsvTranslationDTO csvTranslations) {
-        List<PreContingencyLimitViolationResultDTO> result = self.findNResult(resultUuid, List.of(), Sort.by(Sort.Direction.ASC, ResourceFilterDTO.Column.SUBJECT_ID.getColumnName()));
+        List<PreContingencyLimitViolationResultDTO> result = self.findNResult(resultUuid, List.of(), Sort.by(Sort.Direction.ASC, AbstractLimitViolationEntity.Fields.subjectLimitViolation + SpecificationUtils.FIELD_SEPARATOR + SubjectLimitViolationEntity.Fields.subjectId));
 
         return CsvExportUtils.csvRowsToZippedCsv(csvTranslations.headers(), result.stream().map(r -> r.toCsvRow(csvTranslations.enumValueTranslations())).toList());
     }
 
-    private Sort createNResultSort(Sort sort) {
-        List<Sort.Order> newOrders = new ArrayList<>();
-        sort.forEach(order -> {
-            String property = order.getProperty();
-            if (preContingencyLimitViolationRepository.isParentFilter(property)) {
-                newOrders.add(new Sort.Order(order.getDirection(), "subjectLimitViolation." + property));
-            } else {
-                newOrders.add(order);
-            }
```

Comment:
I think the refactoring broke the children filtering. If I try to use it (with "contains LEM" on the constraint colum) I have the following error : 

```
2024-02-26T10:55:48.923+01:00 ERROR 17043 --- [nio-5023-exec-2] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed: org.springframework.dao.InvalidDataAccessApiUsageException: org.hibernate.query.SemanticException: Could not resolve attribute 'subjectId' of 'org.gridsuite.securityanalysis.server.entities.ContingencyEntity'] with root cause

org.hibernate.query.SemanticException: Could not resolve attribute 'subjectId' of 'org.gridsuite.securityanalysis.server.entities.ContingencyEntity'
	at org.hibernate.query.sqm.SqmPathSource.getSubPathSource(SqmPathSource.java:61) ~[hibernate-core-6.2.6.Final.jar:6.2.6.Final]
	at org.hibernate.query.sqm.tree.domain.AbstractSqmPath.get(AbstractSqmPath.java:187) ~[hibernate-core-6.2.6.Final.jar:6.2.6.Final]
	at org.hibernate.query.sqm.tree.domain.AbstractSqmPath.get(AbstractSqmPath.java:39) ~[hibernate-core-6.2.6.Final.jar:6.2.6.Final]
	at org.gridsuite.securityanalysis.server.repositories.specifications.SpecificationUtils.getColumnPath(SpecificationUtils.java:159) ~[classes/:na]
	at org.gridsuite.securityanalysis.server.repositories.specifications.SpecificationUtils.lambda$contains$8c368af5$1(SpecificationUtils.java:43) ~[classes/:na]
```
	
I didn't review the whole code but I think it is caused by the fact that you removed the kind of property modification that I highlighted above (but I don't say that this is this one specifically).

Reply:
As discussed, the API changed, check unit tests to see how to use the new one